src/backend/access/transam/xact.c

```diff
@@ -298,6 +298,14 @@ static TransactionId unreportedXids[PGPROC_MAX_CACHED_SUBXIDS];
 
 static TransactionState CurrentTransactionState = &TopTransactionStateData;
 
+/*
+ * Hooks for plugins to get control in Transaction Management.
+ */
+TransactionParticipateEnd_hook_type TransactionParticipateEnd_hook = NULL;
```

Comment:
Is is possible to use xact callback here?

Reply:
We can try to handle txn in xact callback as possible.